pkg/generators/cluster.go

```diff
@@ -52,24 +63,59 @@ func (g *ClusterGenerator) GenerateParams(
 		return nil, nil
 	}
 
-	// List all Clusters:
-	clusterSecretList := &corev1.SecretList{}
+	// Do not include the local cluster in the cluster parameters IF there is a non-empty selector
+	// - Since local clusters do not have secrets, they do not have labels to match against
+	ignoreLocalClusters := len(appSetGenerator.Clusters.Selector.MatchExpressions) > 0 || len(appSetGenerator.Clusters.Selector.MatchLabels) > 0
 
-	selector := metav1.AddLabelToSelector(&appSetGenerator.Clusters.Selector, ArgoCDSecretTypeLabel, ArgoCDSecretTypeCluster)
-	secretSelector, err := metav1.LabelSelectorAsSelector(selector)
+	clusters := []argocdappv1.Cluster{}
+
+	// ListCluster from Argo CD's util/db package will include the local cluster in the list of clusters
+	settingsMgr := settings.NewSettingsManager(g.ctx, g.clientset, g.namespace)
+	newDB := db.NewDB(g.namespace, settingsMgr, g.clientset)
+	clustersFromArgoCD, err := newDB.ListClusters(g.ctx)
 	if err != nil {
 		return nil, err
 	}
 
-	if err := g.Client.List(context.Background(), clusterSecretList, client.MatchingLabelsSelector{Selector: secretSelector}); err != nil {
+	if clustersFromArgoCD != nil {
+		clusters = clustersFromArgoCD.Items
+	}
+
+	clusterSecrets, err := g.getSecretsByClusterName(appSetGenerator)
```

Comment:
we can omit this function if `ignoreLocalClusters == false` because this means that there isn't any selector, and we will use all the items from clustersFromArgoCD

Reply:
I believe the `getSecretsByClusterName` is still needed here, as it is used to distinguish local from non-local clusters later in the function body.